codec/types/compat_v2.go

```diff
@@ -0,0 +1,126 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gogo/protobuf/jsonpb"
+	gogoproto "github.com/gogo/protobuf/proto"
+	gogotypes "github.com/gogo/protobuf/types"
+	"google.golang.org/protobuf/encoding/protojson"
+	protov2 "google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+func (any *Any) MarshalJSONPB(m *jsonpb.Marshaler) ([]byte, error) {
+	msg, err := m.AnyResolver.Resolve(any.TypeUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	switch msg.(type) {
+	case protov2.Message:
+		any2 := anypb.Any{
+			TypeUrl: any.TypeUrl,
+			Value:   any.Value,
+		}
+		marshalv2 := protojson.MarshalOptions{
+			Multiline:       false,
+			Indent:          "",
+			AllowPartial:    false,
+			UseProtoNames:   false,
+			UseEnumNumbers:  false,
+			EmitUnpopulated: false,
+			Resolver:        m.AnyResolver.(InterfaceRegistry),
+		}
+		return marshalv2.Marshal(&any2)
+	case gogoproto.Message:
+		gogoany := gogotypes.Any{
+			TypeUrl:              any.TypeUrl,
+			Value:                any.Value,
+			XXX_NoUnkeyedLiteral: any.XXX_NoUnkeyedLiteral,
+			XXX_unrecognized:     any.XXX_unrecognized,
+			XXX_sizecache:        any.XXX_sizecache,
+		}
+		buf := new(bytes.Buffer)
+		err := m.Marshal(buf, &gogoany)
+		return buf.Bytes(), err
+	default:
+		return nil, fmt.Errorf("the message resolved from the Any was not a gogoproto nor a protov2 message, got: %T", msg)
+	}
+}
+
+func (any *Any) UnmarshalJSONPB(u *jsonpb.Unmarshaler, bz []byte) error {
+	if string(bz) == "null" { // we return at this case since JSON will inject "null" for default value fields
```

Comment:
had to add this in, for some reason one of the JSON functions down the line is injecting "null" for  empty `Any` slices ??